tensorflow/core/kernels/gpu_utils.cc

```diff
@@ -37,6 +37,20 @@ using xla::AutotuningLog;
 using xla::ComputeCapability;
 using xla::CudnnVersion;
 
+bool IsBF16NotSupportedInOps(se::Stream *stream) {
+  if (!stream) {
+    return true;  // no stream: don't know whether it's supported
+  }
+#if GOOGLE_CUDA  
+// Performant bfloat16 operations are supported for Ampere+ GPUs. For
+// pre-Ampere GPUs, we cast inputs to float and outputs back to bfloat16.
+  return !stream->GetCudaComputeCapability().IsAtLeast(
+          se::CudaComputeCapability::AMPERE);  
+#elif TENSORFLOW_USE_ROCM
+  return true;  // so far, we return true meaning that the conversion to float is needed
```

Comment:
nit: comments should start with a capital letter and have proper punctuation (i.e. period at the end).